static_frame/test/unit/test_batch.py

```diff
@@ -826,7 +827,325 @@ def test_batch_roll_a(self) -> None:
+        a = [(x[1].shape) for x in f]
+        self.assertTrue(a.count(a[0]) == 3)
+
+    #---------------------------------------------------------------------------
+    def test_batch_relabel_a(self) -> None:
+        f0 = ff.parse('v(int,str,bool,str)|s(9,4)')
+        f1 = Batch.from_frames((
+            f0.iloc[0:3].rename('1'),
+            f0.iloc[3:6].rename('2'),
+            f0.iloc[6:9].rename('3'),
+        )).relabel(columns={2:"two"}).to_frame()
+        actual = f1.columns.values.tolist()
+        expected = [0, 1, 'two', 3]
+        self.assertEqual(expected, actual)
+
+    def test_batch_relabel_b(self) -> None:
+        f0 = ff.parse('v(int,str,bool,str)|s(9,4)')
+        with self.assertRaises(ErrorInitFrame):
+            f1 = Batch.from_frames((
+                f0.iloc[0:2].rename('1'),
+                f0.iloc[2:8].rename('2'),
+            )).relabel(index=("a", "b")).to_frame()
+
+
+    #---------------------------------------------------------------------------
+    def test_batch_relabel_level_add_drop(self) -> None:
+        f0 = ff.parse('v(int,str,bool,str)|s(9,4)')
+        f123 = list(Batch.from_frames((
+            f0.iloc[0:3].rename('1'),
+            f0.iloc[3:6].rename('2'),
+            f0.iloc[6:9].rename('3'),
+        )).relabel_level_add('removeme').items())
+        expected = [['removeme', i] for i in range(9)]
+        actual = [*f123[0][1].index.values.tolist(), *f123[1][1].index.values.tolist(), *f123[2][1].index.values.tolist()]
+        self.assertEqual(expected, actual)
+        f123 = list(Batch.from_frames(np.array(f123)[:,-1]).relabel_level_drop(index=1).items())
+        expected = list(range(9))
+        actual = [*f123[0][1].index.values.tolist(), *f123[1][1].index.values.tolist(), *f123[2][1].index.values.tolist()]
+        self.assertEqual(expected, actual)
+
+
+    def test_batch_relabel_shift_flat(self) -> None:
+        f0 = ff.parse('v(int,str,bool,str)|s(9,4)')
+        f123 = list(Batch.from_frames((
+            f0.iloc[0:3].rename('1'),
+            f0.iloc[3:6].rename('2'),
+            f0.iloc[6:9].rename('3'),
+        )).relabel_shift_in(2, axis=0).relabel_flat(index=1).items())
+        expected = [(i,bool(x)) for i,x in enumerate([1,0,0,1,0,0,1,1,1])]
+        actual = [*f123[0][1].index.values.tolist(), *f123[1][1].index.values.tolist(), *f123[2][1].index.values.tolist()]
+        self.assertEqual(expected, actual)
+
+    #---------------------------------------------------------------------------
+    def test_batch_rank_dense(self) -> None:
+        i = [1,4,7,2,5,8,3,6,9]
+        f0 = Frame.from_items(
+            (
+                ('i',i),
+                ('b',(b%2==0 for b in i))
+            )
+        )
+        f123 = (f0.iloc[0:3],f0.iloc[3:6],f0.iloc[6:9])
+        b123 = Batch.from_frames(f123).rank_dense()
+        for i,(_, f) in enumerate(b123.items()): # type: ignore
+            self.assertTrue(np.array_equal(f.values, f123[i].rank_dense().values))
```

Comment:
This test is basically a property test. If you are interested in learning how to use Hypothesis to implement property tests, let me know. Otherwise, I would just again compare lists, not arrays:


```suggestion
        for i,(_, f) in enumerate(b123.items()): # type: ignore
            self.assertEqual(f.values.tolist(), f123[i].rank_dense().values.tolist())
```